### ISSUE TITLE
nwjs require

### ISSUE THREAD

**Issue by guybedford** [systemjs/systemjs#1161]
Since we changed to no longer altering the `require` global during execution, this breaks Node webkit support as they use a `global.require` there.

We should always scope `require` to be undefined for SystemJS module execution, with synchronous reversion of the global value for AMD and global formats where we need to trigger the right UMD pattern.

Then in terms of consistency, it's annoying that we have different rules for each format like this one, but it does seem best as that is the nature of handling compatibility cases.

**Comment by KenRmk** [comment]
I would very much like to see this fixed as soon as possible since monkey patching my dependencies like this is driving me crazy
```
var isNWJS = typeof nw == 'object' && typeof nw.App == 'object';
if (isNWJS) {
	module.exports = global.require('net');
} else {...}
```

**Comment by KenRmk** [comment]
So I looked into it a bit and found a solution or rather temporal workaround since complexity of SystemJS goes way beyond my understanding. #1168

**Comment by guybedford** [comment]
Unfortunately the `@node` namesapce is reserved only for Node core modules itself. I'd suggest something like the following here rather - 

```javascript
System.set('@nwjs/module-x', System.newModule({ default: require('nwjs-module-x') });
global.require = undefined;
```

Ideally the global require should not be set as that will interfere with global module evaluation.

There is another change that may fix the global require issue here and that would be to move back to using a vm implementation that does encapsulated globals in Node.

Perhaps that is the best fix really, I've created https://github.com/systemjs/systemjs/issues/1183 to track.

**Comment by guybedford** [comment]
I've added clearing the global scoped require in https://github.com/systemjs/systemjs/commit/8ee8916b51c1c2e426acf9ca4193f9f4c31fb82d, pending release.

**Comment by guybedford** [comment]
Released in 0.19.25.
